Foxtrot/Tests/ExtractorTest.cs

```diff
@@ -12,47 +12,262 @@
+                    libPaths: new[] { @"Microsoft.Research\RegressionTest\ClousotTestHarness\bin\debug" },
+                    compilerCode: "",
+                    useBinDir: false,
+                    useExe: false,
+                    mustSucceed: false);
+                yield return new Options(
+                    sourceFile: @"Foxtrot\Tests\CheckerTests\CC1016.cs",
+                    foxtrotOptions: null,
+                    useContractReferenceAssemblies: true,
+                    compilerOptions: null,
+                    references: new[] { "mscorlib.dll", "System.dll", "ClousotTestHarness.dll" },
+                    libPaths: new[] { @"Microsoft.Research\RegressionTest\ClousotTestHarness\bin\debug" },
+                    compilerCode: "CS",
+                    useBinDir: false,
+                    useExe: false,
+                    mustSucceed: false);
+                yield return new Options(
+                    sourceFile: @"Foxtrot\Tests\CheckerTests\CheckerErrors.cs",
+                    foxtrotOptions: null,
+                    useContractReferenceAssemblies: true,
+                    compilerOptions: null,
+                    references: new[] { "mscorlib.dll", "System.dll", "ClousotTestHarness.dll" },
+                    libPaths: new[] { @"Microsoft.Research\RegressionTest\ClousotTestHarness\bin\debug" },
+                    compilerCode: "CS",
+                    useBinDir: false,
+                    useExe: false,
+                    mustSucceed: false);
+            }
+        }
+
+        public static IEnumerable<object> ExtractorTest
+        {
+            get
             {
-                try
-                {
-                    string sourceFile = (string)TestContext.DataRow["Name"];
-                    string options = (string)TestContext.DataRow["Options"];
-                    TestContext.WriteLine("FAILED: Name={0} Options={1}", sourceFile, options);
-                }
-                catch { }
+                return Enumerable.Range(0, ExtractorTestData.Count()).Select(i => new object[] { i });
             }
         }
 
-        [DeploymentItem("Foxtrot\\Tests\\TestInputs.xml"), DataSource("Microsoft.VisualStudio.TestTools.DataSource.XML", "|DataDirectory|\\TestInputs.xml", "ExtractorTest", DataAccessMethod.Sequential)]
-        [TestMethod]
-        [TestCategory("Runtime"), TestCategory("CoreTest"), TestCategory("V4.0"), TestCategory("Short")]
-        public void ExtractorFailures()
+        [Theory]
+        [MemberData("ExtractorTest")]
+        [Trait("Category", "Runtime")]
+        [Trait("Category", "CoreTest")]
+        [Trait("Category", "V4.0")]
+        [Trait("Category", "Short")]
+        public void ExtractorFailures(int testIndex)
         {
-            var options = new Options(this.TestContext);
+            Options options = ExtractorTestData.ElementAt(testIndex);
```

Comment:
As I mentioned above, the problem is the `Options` class is exceptionally poorly designed as a storage location for test inputs. The complexity is due exclusively to this design and would be corrected in the future if `Options` were made immutable or if we passed parameter values directly to the test methods without wrapping them in `Options`.